src/ape/api/providers.py

```diff
@@ -143,12 +143,11 @@ def disconnect(self):
         Disconnect from a provider, such as tear-down a process or quit an HTTP session.
         """
 
-    @property
+    @abstractmethod
```

Comment:
- [ ] Change places where we catch `ProviderNotConnectedError` to first checking `is_connected()` instead.